### ISSUE TITLE
sync client: Object functions fail after ~60 minutes

### ISSUE THREAD

**Issue by compressed** [ThouCheese/cloud-storage-rs#34]
After about 60 minutes after starting my service, I always run into the following error:

```
error: Reqwest(reqwest::Error { kind: Request, url: "https://www.googleapis.com/oauth2/v4/token", source: hyper::Error(Connect, ConnectError("dns error", ResolveError { kind: Proto(ProtoError { kind: Message("could not send request"), backtrack: None })
```

I am doing `Object::download_sync` at this time. I think this coincides with the initial token expiration. I don't think the dns request is failing (running on GKE), but struggling to find out where the error actually is...

Running version 0.6.

**Comment by compressed** [comment]
I guess it is coming from trust-dns:

https://github.com/bluejekyll/trust-dns/blob/fb5716cfb5a238eb3d920b4aa655bccbce497242/crates/proto/src/xfer/mod.rs#L243

Going to try without that reqwest feature.

**Comment by ThouCheese** [comment]
Hi! Thanks for your bug report. We're you able to confirm that this issue is caused by trust-dns?

**Comment by compressed** [comment]
Sure. Yep, it seems to go away by removing trust-dns. Running for 2+ hrs now no issues. Maybe we can make that an optional setting? Or just not use it for now.. I don't know too much about the stability of that crate.

**Comment by ThouCheese** [comment]
This is now fixed with version `0.6.1`, which has trust-dns under a feature flag which is disabled by default.
